data_layer/avail/utils.go

```diff
@@ -11,6 +12,10 @@ func writeConfigToTOML(path string, c Avail) error {
 	if err != nil {
 		return err
 	}
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
```

Comment:
why executable? 0644 should be enough

Reply:
The execution permission is necessary to allow files creation within that directory